clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslServer.java

```diff
@@ -53,14 +54,22 @@
     private static final String NEGOTIATED_PROPERTY_KEY_TOKEN = OAuthBearerLoginModule.OAUTHBEARER_MECHANISM + ".token";
     private static final String INTERNAL_ERROR_ON_SERVER = "Authentication could not be performed due to an internal error on the server";
     private static final String SASLNAME = "(?:[\\x01-\\x7F&&[^=,]]|=2C|=3D)+";
+    private static final String KEY = "[A-Za-z]+";
+    private static final String VALUE = "[\\x01-\\x7F]+";
```

Comment:
`ScramMessages`'s value regex is `"[\\x01-\\x7F&&[^,]]+"`. I tried using it but could not get my tests to pass. 
To be frank, I don't understand it at all. Specifically the `&&[^,]` part.

[Regex101](https://regex101.com/r/ED5kuX/1) says 
> &&[^, matches a single character in the list &[^, (case sensitive)
but I doubt that that is the case, otherwise I think SCRAM would not work as well.
Is this some feature in the Java Regex engine I'm not familiar with?